packages/gatsby/src/redux/reducers/queries.ts

```diff
@@ -190,7 +190,16 @@ export function queriesReducer(
       const { path } = action.payload
       const query = registerQuery(state, path)
       query.dirty = 0
-      query.running = 0 // TODO: also reset the state on pre-bootstrap (as it could've been persisted)
+      query.running = 0 // TODO: also
+      return state
+    }
+    case `SET_PROGRAM_STATUS`: {
+      if (action.payload === `BOOTSTRAP_FINISHED`) {
+        // Reset the running state (as it could've been persisted)
+        for (const [, query] of state.trackedQueries) {
+          query.running = 0
+        }
```

Comment:
Hooked into onPostBootstrap as couldn't find an easy way to hook into onPreBootstrap. It should still wipe out the `running` state before we start the develop server or run any queries.